synapse/rest/client/capabilities.py

```diff
@@ -72,6 +76,11 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
                 "org.matrix.msc3244.room_capabilities"
             ] = MSC3244_CAPABILITIES
 
+        if self.config.experimental.msc2716_enabled:
+            response["capabilities"]["m.room_versions"][
+                "org.matrix.msc2716.room_capabilities"
```

Comment:
this should not be under a separate `org.matrix.msc2716` key: there should be a single `room_capabilities` key (or, until msc3244 lands, `org.matrix.msc3244.room_capabilities`, which could have unstable keys under it.

Quite honestly it makes me queasy to build more unstable features on top of existing unstable features like this. What's to stop us finalising MSC3244 first?

Reply:
Ohh, I think I misunderstood the `org.matrix.msc3244.room_capabilities` field. I didn't look into what [MSC3244](https://github.com/matrix-org/matrix-doc/pull/3244) actually was and thought it was describing features of that MSC like `knock` although with hindisght, doesn't make sense for a single MSC to have `knock` and `restricted` in one 😁

Will nest under the unstable `room_capabilities` field 👍

Reply:
Done ✅

`GET http://localhost:8008/_matrix/client/v3/capabilities`
```json
{
    "capabilities": {
        "m.room_versions": {
            "default": "9",
            "available": {
                "1": "stable",
                "2": "stable",
                "3": "stable",
                "4": "stable",
                "5": "stable",
                "6": "stable",
                "7": "stable",
                "8": "stable",
                "9": "stable",
                "org.matrix.msc2176": "unstable",
                "org.matrix.msc2716v3": "unstable"
            },
            "org.matrix.msc3244.room_capabilities": {
                "knock": {
                    "preferred": "7",
                    "support": [
                        "7",
                        "8",
                        "9",
                        "org.matrix.msc2716v3"
                    ]
                },
                "restricted": {
                    "preferred": "9",
                    "support": [
                        "8",
                        "9"
                    ]
                },
                "org.matrix.msc2716.historical": {
                    "preferred": "org.matrix.msc2716v3",
                    "support": [
                        "org.matrix.msc2716v3"
                    ]
                }
            }
        },
        "m.change_password": {
            "enabled": true
        },
        "m.set_displayname": {
            "enabled": true
        },
        "m.set_avatar_url": {
            "enabled": true
        },
        "m.3pid_changes": {
            "enabled": true
        }
    }
}
```